### ISSUE TITLE
push subcommand is failed when a commit message includes single quote or double quote

### ISSUE THREAD

**Issue by t2y** [kazamori/backlog-github-integration-action#6]
There are bugs about invoking docker run with JSON data when a commit message has quoted string.

### single quote

```
    \"message\": \"include 'single quoted text' message\",
```

```bash
Unmatched arguments from index 7: 'quoted', 'text message",
    "timestamp": "2022-05-10T19:20:31+09:00",
    "tree_id": "e05aa82f487db9f80f9d020e023cda7233f4d221",
    "url": "https://github.com/kazamori/REPO/commit/d42e7ee114bfa76932f3e2e[53](https://github.com/kazamori/REPO/runs/6368060156?check_suite_focus=true#step:8:53)29f75f730dc9b2a"
  }
]'
Usage: Backlog GitHub integration action push --commits=<commits>
       --pusher=<pusher> --repository=<repository>
push event that trigger workflows
      --commits=<commits>   set commits in the event
      --pusher=<pusher>     set pusher in the event
      --repository=<repository>
                            set repository (owner/repo) in the event
```

### double quote

```
    \"message\": \"include \\"quoted text\\" message\",
```

```bash
Missing required options: '--repository=<repository>', '--pusher=<pusher>', '--commits=<commits>'
Usage: Backlog GitHub integration action push --commits=<commits>
       --pusher=<pusher> --repository=<repository>
push event that trigger workflows
      --commits=<commits>   set commits in the event
      --pusher=<pusher>     set pusher in the event
      --repository=<repository>
                            set repository (owner/repo) in the event
```

**Comment by t2y** [comment]
The push subcommand cannot handle a single quote in JSON since JSON is quoted by a single quote.

```
... --commits '[ {...}, {...}, {...}]' : whole quoted strings are JSON data
```

But, for example, handling some strings on bash.

```
... --commits '[ {...}, {"message": "aaa ' bbb ' ccc"}, {...}]'
```

As multiple strings are handled.

1. : '[ {...}, {"message": "aaa '
1. : bbb 
1. : ' ccc"}, {...}]'

A workaround is escaping single quotes before passing arguments to the push subcommand.

```bash
ESCAPED=$(echo ${JSON_DATA} | sed "s/'//g")
```

**Comment by t2y** [comment]
In double-quoted string case, it seems picocli handles and occurred an error.

```
Missing required options: ...
```

* https://picocli.info/#_quoted_values

**Comment by t2y** [comment]
Hmm, double-quoted string case, this is a bug of arguments splitter. But, the splitter behavior has complicated requirements.

**Comment by t2y** [comment]
It handled unexpected arguments length when I included double-quoted strings.

```
    \"message\": \"include \\"double quoted text\\" message\",
```

I confirmed to output logs about the argument length the push subcommand got.

```
[main] DEBUG jp.kazamori.github.actions.backlog.Main - arguments length: 4
```

**Comment by t2y** [comment]
I thought action.yml has two args for inputs, then two arguments will be passed. But, it's not always so when escaping quoted strings. I was wrong.

```yml
  args:
    - ${{ inputs.subcommand }}
    - ${{ inputs.args }}
```

```java
    static String[] ensureArgumentTokens(String[] args) {
        // for GitHub Actions
        // it takes inputs as two strings like below
        // "pull_request" "--repository kazamori/backlog-github-integration-action --pr-number 1"
        if (args.length != 2) {
            return args;
        }
        val tokens = StrUtil.extractStringToTokens(args[0], args[1]);
        tokens.add(0, args[0]);
        return tokens.toArray(new String[]{});
    }
```

**Comment by t2y** [comment]
About single-quoted strings are the same, too. Main can get any arguments length and parse before converting tokens for picocli.

**Comment by t2y** [comment]
This can be processed without an error. But, it's not expected.

```
    \"message\": \"***-3 include \\\\"double quoted text\\\\" message\",
```

then, backlog comment.

```
include \double quoted text\ message (fe6925)
```

**Comment by t2y** [comment]
The single-quoted strings are valid. It's completed. :rocket: 

```
    \"message\": \"include 'single quoted text' message\",
```

**Comment by t2y** [comment]
### Summary

* single-quoted: completely fix
* double-quoted: only handle `\\\\"` strings, and double-quote character would be erased in the message

**Comment by t2y** [comment]
After confirming no regression in my production environment, I will merge this fix to the v1 branch.

**Comment by t2y** [comment]
I released these changes as v1.0.1.
